zboxcore/sdk/common_test.go

```diff
@@ -0,0 +1,393 @@
+package sdk
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strings"
+	"sync"
+	"testing"
+
+	"github.com/0chain/gosdk/zboxcore/blockchain"
+	"github.com/0chain/gosdk/zboxcore/sdk/mocks"
+	"github.com/stretchr/testify/assert"
+
+	"gopkg.in/yaml.v2"
+)
+
+const (
+	configDir            = "testdata"
+	syncTestDir          = configDir + "/" + "sync"
+	syncDir              = syncTestDir + "/" + "sync_alloc"
+	textPlainContentType = "text/plain"
+)
+
+func blobberIDMask(idx int) string {
+	return fmt.Sprintf("${blobber_id_%v}", idx)
+}
+
+func blobberURLMask(idx int) string {
+	return fmt.Sprintf("${blobber_url_%v}", idx)
+}
+
+func parseFileContent(t *testing.T, fileName string, jsonUnmarshalerInterface interface{}) (fileContentBytes []byte) {
+	fs, err := os.OpenFile(fileName, os.O_RDONLY, 0644)
+	assert.NoErrorf(t, err, "Error os.OpenFile() %v: %v", fileName, err)
+
+	defer fs.Close()
+	bytes, err := ioutil.ReadAll(fs)
+	assert.NoErrorf(t, err, "Error ioutil.ReadAll() cannot read file content of %v: %v", fileName, err)
+	if jsonUnmarshalerInterface != nil {
+		err = json.Unmarshal(bytes, jsonUnmarshalerInterface)
+		assert.NoErrorf(t, err, "Error json.Unmarshal() cannot parse file content to %T object: %v", jsonUnmarshalerInterface, err)
+	}
+
+	return bytes
+}
+
+func writeFileContent(t *testing.T, fileName string, fileContentBytes []byte) {
+	fs, err := os.OpenFile(fileName, os.O_RDWR|os.O_CREATE, 0644)
+	assert.NoErrorf(t, err, "Error os.OpenFile() %v: %v", fileName, err)
+	defer fs.Close()
+	_, err = fs.Write(fileContentBytes)
+	assert.NoErrorf(t, err, "Error fs.Write() cannot write file content to %v: %v", fileName, err)
+}
+
+type nodeConfig struct {
```

Comment:
Why is this object needed. It only seems to be used to to define four objects, can't you set them directory. You could pass this to `setupMockInitStorageSDK` directly from the unit test if the values are going to change.